### PR TITLE
[SPARK-28366][CORE][FOLLOW-UP] Refine logging in driver when loading single large unsplittable file

### PR DESCRIPTION
**Pull request by WeichenXu123**
## What changes were proposed in this pull request?

* Add log in `NewHadoopRDD`
* Remove some words in logs which related to specific user API.

## How was this patch tested?

Manual.

Please review https://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala [hunk 1]
```diff
@@ -215,8 +215,7 @@ class HadoopRDD[K, V](
           val codecFactory = new CompressionCodecFactory(jobConf)
           if (Utils.isFileSplittable(path, codecFactory)) {
             logWarning(s"Loading one large file ${path.toString} with only one partition, " +
-              s"we can increase partition numbers by the `minPartitions` argument in method " +
-              "`sc.textFile`")
+              s"we can increase partition numbers for improving performance.")
```
Comment (HyukjinKwon) [inline core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala]: I am okay with it

**Comment by HyukjinKwon** [comment]
I am okay with this logging stuff though don't feel strongly about it. Let me leave it to @cloud-fan.

**Comment by WeichenXu123** [comment]
CC @cloud-fan

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**